src/code.cpp

```diff
@@ -0,0 +1,22 @@
+#include <cpp11.hpp>
+#include <vector>
+using namespace cpp11;
+
+[[cpp11::register]]
+cpp11::writable::integers which_within_dist(doubles_matrix<> distmat, doubles idx, double dist) {
+
+  int n_idx = idx.size();
+  int n_matrix = distmat.ncol();
+  std::vector<int> out;
+
+  for (int i = 0; i < n_idx; i++) {
+    for (int j = 0; j < n_matrix; j++) {
+      if (distmat(idx[i] - 1, j) <= dist) {
```

Comment:
Hee I would suggest that you moving defining int id = idx[i] - 1 instead of doing it inside the second loop.

```suggestion
    int id = idx[i] - 1;
    for (int j = 0; j < n_matrix; j++) {
      if (distmat(id, j) <= dist) {
```